flytestdlib/otelutils/factory.go

```diff
@@ -30,14 +33,15 @@ const (
 )
 
 var tracerProviders = make(map[string]*trace.TracerProvider)
-var noopTracerProvider = rawtrace.NewNoopTracerProvider()
+var noopTracerProvider = noop.NewTracerProvider()
 
 func RegisterTracerProvider(serviceName string, config *Config) error {
```

Comment:
@Sovietaced I ended up not taking a `context` param here to avoid a breaking change

Reply:
Maybe there should be another function that takes in a context. Otherwise you won't be able to gracefully shut down the OTLP exporters and the server may shut down with buffered data points.

Reply:
Added. Thanks for the suggestion.